--- orientdb-community-2.0/bin/shutdown.sh	2015-01-20 16:41:19.000000000 +0000
+++ bin-orientdb/shutdown.sh	2015-01-30 22:25:12.243263000 +0000
@@ -17,30 +17,14 @@
   esac
 fi
 
-while [ -h "$PRG" ]; do
-  ls=`ls -ld "$PRG"`
-  link=`expr "$ls" : '.*-> \(.*\)$'`
-  if expr "$link" : '/.*' > /dev/null; then
-    PRG="$link"
-  else
-    PRG=`dirname "$PRG"`/"$link"
-  fi
-done
 
-# Get standard environment variables
-PRGDIR=`dirname "$PRG"`
+# The original had logic here to find the 'ORIENTDB_HOME' path. In the
+# Conveyor setup, that's clearly specified so we've cut it all
+# out. These runtime settings never need change.
+ORIENTDB_HOME="$HOME/.conveyor/runtime/dogfoodsoftware.com/conveyor-orientdb"
+CONFIG_FILE="$ORIENTDB_HOME/config/orientdb-server-config.xml"
+LOG_FILE="$ORIENTDB_HOME/config/orientdb-server-log.properties"
 
-# Only set ORIENTDB_HOME if not already set
-[ -f "$ORIENTDB_HOME"/bin/server.sh ] || ORIENTDB_HOME=`cd "$PRGDIR/.." ; pwd`
-export ORIENTDB_HOME
-cd "$ORIENTDB_HOME/bin"
-
-if [ ! -f "${CONFIG_FILE}" ]
-then
-  CONFIG_FILE=$ORIENTDB_HOME/config/orientdb-server-config.xml
-fi
-
-# Set JavaHome if it exists
 if [ -f "${JAVA_HOME}/bin/java" ]; then 
    JAVA=${JAVA_HOME}/bin/java
 else
@@ -48,7 +32,6 @@
 fi
 export JAVA
 
-LOG_FILE=$ORIENTDB_HOME/config/orientdb-server-log.properties
 LOG_LEVEL=warning
 WWW_PATH=$ORIENTDB_HOME/www
 JAVA_OPTS=-Djava.awt.headless=true
